plugins/inputs/haproxy/haproxy_test.go

```diff
@@ -134,17 +134,19 @@ func TestHaproxyGeneratesMetricsUsingSocket(t *testing.T) {
 	var randomNumber int64
 	var sockets [5]net.Listener
 
-	_globmask := filepath.Join(os.TempDir(), "test-haproxy*.sock")
-	_badmask := filepath.Join(os.TempDir(), "test-fail-haproxy*.sock")
+	// The Maximum length of the socket path is 104/108 characters, path created with t.TempDir() is too long for some cases
+	// (it combines test name with subtest name and some random numbers in the path)
+	//nolint:usetesting // "os.TempDir() could be replaced by t.TempDir() in TestHaproxyGeneratesMetricsUsingSocket"
```

Comment:
Hmmm this comment is misleading isn't it as we cannot replace the call here can we?

Reply:
Wording changed.